mlserver/codecs/utils.py

```diff
@@ -59,12 +60,14 @@ def decode_request_input(
 
 
 def decode_inference_request(
-    inference_request: InferenceRequest, metadata_inputs: Dict[str, MetadataTensor] = {}
+    inference_request: InferenceRequest,
+    model_settings: ModelSettings = None,
+    metadata_inputs: Dict[str, MetadataTensor] = {},
```

Comment:
where is that used?

Reply:
There's a `self.decode_request()` method in the `MLModel` base class which calls this one. Initially, there was also a middleware layer which would call this method, although we scrapped that idea for now.